pkg/ttnpb/errors.go

```diff
@@ -62,14 +81,21 @@ func ErrorDetailsToProto(e errors.ErrorDetails) *ErrorDetails {
 		if err != nil {
 			panic(fmt.Sprintf("Failed to encode error attributes: %s", err)) // Likely a bug in ttn (invalid attribute type).
 		}
-		proto.Attributes = attributesStruct
+		pb.Attributes = attributesStruct
 	}
 	if cause := e.Cause(); cause != nil {
 		if ttnErr, ok := errors.From(cause); ok {
-			proto.Cause = ErrorDetailsToProto(ttnErr)
+			pb.Cause = ErrorDetailsToProto(ttnErr)
+		}
+	}
+	for _, d := range e.Details() {
+		dAny, err := types.MarshalAny(d)
+		if err != nil {
+			panic(fmt.Sprintf("Failed to encode error details: %s", err))
 		}
+		pb.Details = append(pb.Details, dAny)
```

Comment:
This, together with 

https://github.com/TheThingsNetwork/lorawan-stack/blob/2c3d0873430c542c0c24c0bca7dede7edfb766ed/pkg/errors/grpc.go#L104-L108

and

https://github.com/TheThingsNetwork/lorawan-stack/blob/master/pkg/errors/grpc.go#L116

results in details being added twice to error responses. We shouldn't duplicate things like that. We should either communicate additional error detail protos with gRPC's error details or put them in our own ErrorDetails. Not both (unless we're going to separate them on package, so that `google.rpc.*` protos go into gRPC error details, and `ttn.lorawan.v3` protos go into TTN ErrorDetails).